WITH_LCM = OFF leads to build error

When LCM support is disabled, `drakeRBSystem` is not built (see [this line](https://github.com/RobotLocomotion/drake/blob/master/drake/systems/plants/CMakeLists.txt#L144)), `drakeRBSystem_export.h` is not generated, and compilation of `RigidBodyTree.cpp` fails with

```
In file included from <path-to-drake-distro>/drake/systems/plants/RigidBodyTree.cpp:3:
In file included from <path-to-drake-distro>/drake/../drake/systems/plants/RigidBodyTree.h:17:
<path-to-drake-distro>/drake/../drake/systems/plants/RigidBodyFrame.h:6:10: fatal error: 'drake/drakeRBSystem_export.h' file not found
#include "drake/drakeRBSystem_export.h"
```

This error occurs because `drakeRBSystem` is not currently listed as a dependency for `drakeRBM`. I think we should keep it that way and eliminate the need to include `drakeRBSystem_export.h` in `RigidBodyFrame.h`.

I also just accidentally encountered this issue on my machine, except it's not because I set `WITH_LCM = OFF`. Instead, for some reason, LCM was not found on my machine when I tried to compile Drake this morning:

```
-- checking for module 'lcm'
--   package 'lcm' not found
-- disabling LCM.  lcm types will not be built.
```
Later, as this issue states, there's a compiler error due to a missing `drakeRBSystem_export.h`.
```
[ 26%] Building CXX object systems/plants/CMakeFiles/drakeRBM.dir/RigidBodyTree.cpp.o
In file included from /home/liang/dev/drake/drake/../drake/systems/plants/RigidBodyTree.h:15:0,
                 from /home/liang/dev/drake/drake/systems/plants/RigidBodyTree.cpp:1:
/home/liang/dev/drake/drake/../drake/systems/plants/RigidBodyFrame.h:5:40: fatal error: drake/drakeRBSystem_export.h: No such file or directory
 #include "drake/drakeRBSystem_export.h"
```

--- a/drake/systems/plants/RigidBodyFrame.h
+++ b/drake/systems/plants/RigidBodyFrame.h
@@ -2,7 +2,7 @@
 
 #include <Eigen/Dense>
 
-#include "drake/drakeRBSystem_export.h"
+#include "drake/drakeRBM_export.h"
 #include "drake/systems/plants/RigidBody.h"
 
 namespace tinyxml2 {

